### PR TITLE
added lilygo t-display (esp32-based) board definition

### PR DESCRIPTION
**Pull request by PBrunot**
## Description of Change

This change adds a new board lilygo-t-display based on esp32. Board variants have 4 MB or 16 MB flash memory. All boards come with a TFT display. This board is the predecessor to the already included lilygo-t-display-s3.

[Manufacturer Board webpage](https://www.lilygo.cc/products/lilygo%C2%AE-ttgo-t-display-1-14-inch-lcd-esp32-control-board)

[Pinmap](https://github.com/Xinyuan-LilyGO/TTGO-T-Display/blob/master/image/pinmap.jpg)

## Tests scenarios

I have tested my pull request with a Lilygo TDisplay HW rev 1.1 by configuring PlatformIO to use the platform in https://github.com/PBrunot/arduino-esp32#feature-lilygo-tdisplay as framework for arduino-esp32, and uploading a sketch to a destination board successfully. Testing sketch is using serial, TFT display over SPI.

platformio.used for testing
```
[env:T-Display]
platform = https://github.com/platformio/platform-espressif32.git
board = lilygo_t-display
framework = arduino
monitor_speed = 115200
build_flags = -D ARDUINO_LILYGO_T_DISPLAY_S3
platform_packages =
    framework-arduinoespressif32 @ https://github.com/PBrunot/arduino-esp32#feature-lilygo-tdisplay
lib_deps = 
	moononournation/GFX Library for Arduino@^1.3.7
	Wire
	SPI
```

## Related links

This is required to add support in platformio for the same board, as per platformio esp32 policies : https://github.com/platformio/platform-espressif32/issues/209

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -2958,6 +2958,162 @@ S_ODI_Ultra.menu.EraseFlash.all.upload.erase_cmd=-e
...
+lilygo_t_display.menu.FlashFreq.40.build.flash_freq=40m
+
+lilygo_t_display.menu.FlashSize.4M=4MB (32Mb)
+lilygo_t_display.menu.FlashSize.4M.build.flash_size=4MB
+lilygo_t_display.menu.FlashSize.8M=8MB (64Mb)
+lilygo_t_display.menu.FlashSize.8M.build.flash_size=8MB
+lilygo_t_display.menu.FlashSize.8M.build.partitions=default_8MB
+lilygo_t_display.menu.FlashSize.2M=2MB (16Mb)
+lilygo_t_display.menu.FlashSize.2M.build.flash_size=2MB
+lilygo_t_display.menu.FlashSize.2M.build.partitions=minimal
```
Comment (P-R-O-C-H-Y) [inline boards.txt:3083]: As the boards will be 4MB or 16MB, this part can be removed and users will not be able to select wrong flash size.

File: boards.txt [hunk 2]
```diff
@@ -2958,6 +2958,162 @@ S_ODI_Ultra.menu.EraseFlash.all.upload.erase_cmd=-e
...
+lilygo_t_display.menu.DebugLevel.error.build.code_debug=1
+lilygo_t_display.menu.DebugLevel.warn=Warn
+lilygo_t_display.menu.DebugLevel.warn.build.code_debug=2
+lilygo_t_display.menu.DebugLevel.info=Info
+lilygo_t_display.menu.DebugLevel.info.build.code_debug=3
+lilygo_t_display.menu.DebugLevel.debug=Debug
+lilygo_t_display.menu.DebugLevel.debug.build.code_debug=4
+lilygo_t_display.menu.DebugLevel.verbose=Verbose
+lilygo_t_display.menu.DebugLevel.verbose.build.code_debug=5
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:3114]: Please add EraseFlash at the end of board definition. Else this board will not work in Arduino IDE.
```
lilygo_t_display.menu.EraseFlash.none=Disabled
lilygo_t_display.menu.EraseFlash.none.upload.erase_cmd=
lilygo_t_display.menu.EraseFlash.all=Enabled
lilygo_t_display.menu.EraseFlash.all.upload.erase_cmd=-e
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@PBrunot Please take a look on my comments. After its fixed the board addition can be merged :)

**Comment by PBrunot** [comment]
Thanks, I've implemented your comments.

**Comment by SuGlider** [comment]
Attention::: WRONG FILE NAME ---> [pins_arduino.h.txt](https://github.com/espressif/arduino-esp32/pull/8533/files#diff-ec7cb232a3c534c4ec6f10b4d5143ad9d6c87760985f277d46370ad3d5aa0211)

I'll fix it in the GPIO48 PR.

**Comment by PBrunot** [comment]
Sorry about that & thank you for your fix.

**Pull request merged**